### ISSUE TITLE
Unable to customize spinner color via theme

### ISSUE THREAD

**Issue by ekimlinger** [GeekyAnts/NativeBase#1624]
## react-native, react and native-base version
```
"native-base": "^2.3.5",
"react-native": "0.53.3",
```
## Expected behaviour
`color` should be a style prop that we should be able to pass in after injecting, from our `native-base-theme/components/Spinner.js` file.
## Actual behaviour
`Warning: Failed prop type: Invalid props.style key 'color' supplied to 'Spinner'.`
## Steps to reproduce (code snippet or screenshot)
Spinner.js:
```
import variable from './../variables/platform';

export default (variables = variable) => {
    const spinnerTheme = {
        '.primary': {
            color: variables.primaryColor,
        },
       ...more customizations...
        height: 80,
    };

    return spinnerTheme;
};
```
## Is the bug present in both ios and android or in any one of them?
Both
## Any other additional info which would help us debug the issue quicker.
I do realize that `color` is a prop that gets passed down to the underlying `ActivityIndicator` component from react-native, however we should be able to customize this via the nativebase style provider.

**Comment by akhil-geekyants** [comment]
@ekimlinger you can set the default spinner color using `defaultSpinnerColor` variable in native-base-theme/variables/*.js. See Gif

Gif 

![spinner](https://user-images.githubusercontent.com/27768184/36655983-0465b5b8-1aec-11e8-8f07-aa07f254ae33.gif)

**Comment by ekimlinger** [comment]
My apologies @akhil-geekyants, I neglected to mention that I have already set the `defaultSpinnerColor` in my app's theme. 
This request is for the ability to customize the spinner based on a class name or dynamic styles. If the underlying react-native component uses color as a style property, we should be destructuring that off the `style` that is given to us by `connectStyle` and then forwarding that down.
